### ISSUE TITLE
Issue with indentation level of pypi installed package

### ISSUE THREAD

**Issue by vsoch** [prefix-dev/pixi#557]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

I tried to create a Dockerfile to reproduce what was going on:

```
FROM fedora

# docker build -t test .
# docker run -it test
# . ~/.bash_profile
# pixi shell
# ipython 
# from google.cloud.storage.batch import *

RUN /bin/bash -c "curl -fsSL https://pixi.sh/install.sh | bash" && \
    . ~/.bashrc && \
    . ~/.bash_profile && \
    pixi init pixi-env && \
    cd pixi-env && \
    pixi add python && \
    pixi add --pypi google-cloud-storage && \
    pixi add IPython
WORKDIR /pixi-env
```
But it didn't! So all I can do is show you what is happening on my local machine (and maybe others can identify or reproduce)? On my machine, it told me the indentation level was incorrect, and I can only guess it grabbed the wrong package for my distribution with some weird newline or character?

```
-testing-llnl --googlebatch-snippets intel-mpi
IndentationError in file /home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/site-packages/google/cloud/storage/batch.py, line 325:
unindent does not match any outer indentation level (batch.py, line 325)
  File "/home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/site-packages/smart_open/__init__.py", line 34, in <module>
  File "/home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/site-packages/smart_open/smart_open_lib.py", line 35, in <module>
  File "/home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/site-packages/smart_open/doctools.py", line 21, in <module>
  File "/home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/site-packages/smart_open/transport.py", line 101, in <module>
  File "/home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/site-packages/smart_open/transport.py", line 49, in register_transport
  File "/home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/importlib/__init__.py", line 90, in import_module
  File "/home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/site-packages/smart_open/gcs.py", line 15, in <module>
  File "/home/sochat1/Desktop/Code/snek/snakemake-executor-plugin-googlebatch/.pixi/env/lib/python3.12/site-packages/google/cloud/storage/__init__.py", line 39, in <module>
```
I was able to fix this with pip uninstall (the pip in my pixi environment) and then pip install of the same package. I'll follow up if I can reproduce it.

### Issue description

Indentation level is incorrect when installing pypi package (see above).

### Expected behavior

I would expect the install to work without issue. Thank you!

**Comment by baszalmstra** [comment]
Thats really weird! thanks for reporting! We’ll take a look at it!

**Comment by tdejager** [comment]
No luck reproducing either. Thank you for the detailed write-up though.

Do you have the lock file resolved? You could try the `--no-install` flag for the add comand. Then we could check what `.whl` it is actually using.

**Comment by vsoch** [comment]
I can try this next time!

An unrelated (?) note - when I tried rebuilding a previously working pixi.toml that had pytorch libraries and keras and python, it wouldn't resolve, and then when I changed the order (to add python last) it would take maybe 5-10 per resolve, even for simple packages. It felt like a bug but I'm not sure (and I saw it was falling back to conda). At that point I pulled an old container and grabbed the pixi.lock to avoid it, but if I bring that container up again I can give you the exact definition to reproduce.
